shared/logger/native-logger.js

```diff
@@ -1,8 +1,30 @@
 // @flow
 import {log, dump} from '../native/logger'
-import type {Logger, LogLevel} from './types'
+import type {Logger, LogLine, LogLevel} from './types'
 import {toStringForLog} from '../util/string'
 
+const parseLine = (l: string): LogLine => {
```

Comment:
how much work would it be for this to just store the object and make the string when dumping. it's faster (like how the old logger worked) and doesn't have this problem at all

Reply:
i'm talking about the ring buffers. we should talk about the flows
i hope we're not
1. building a string
2. sending it over to native
3. reading it back or something
4. sending it back to native

Reply:
This is the NativeLogger, so there are no ring buffers involved. But the log flow is (see `log` function in this file):

1. build a string `toLog`
2. stringify `[Date.now(), toLog]`
3. send to native (via native `log` function)

This function is called during the `logSend` flow, where we:

1. Destringify each line received from native into `[ts, line]`
2. Eventually stringify `[level, ts, line]` into a file
3. Upload that file